barretenberg/cpp/pil/vm2/alu.pil

```diff
@@ -123,6 +138,83 @@ sel_op_add * (1 - sel_op_add) = 0;
 #[ALU_ADD]
 sel_op_add * (1 - sel_tag_err) * (ia + ib - ic - cf * (max_value + 1)) = 0;
 
+// MUL
+
+sel_op_mul * (1 - sel_op_mul) = 0;
+
+pol commit c_hi;
+
+// MUL - non u128
+
+// TODO(MW): cf required? could be sel_op_mul * (1 - sel_tag_err) * (ia * ib - ic - cf * (max_value + 1) * c_hi) = 0;
+#[ALU_MUL_NON_U128]
+sel_op_mul * IS_NOT_U128 * (1 - sel_tag_err)
+    * (
+        ia * ib
+        - ic
+        - (max_value + 1) * c_hi
+    ) = 0;
```

Comment:
Since in the non-u128 case we can never overflow the field with `a*b`, I don't think we need to use `cf` here (c_hi will be 0 anyway when we overflow the tag's bits), but wanted to double check?

Reply:
My initial thought is no because it's not possible to set `c_hi` to anything that can over/underflow since it must must `a*b - c` where all values are tag constrained to certain sizes. i.e. I don't think for bit size `n` we can set `c_hi` such that it passes `= (a*b - c) / 2^n` unless it's the 'good' value.
This isn't true for `u128` but we have gated that case